syntax = "proto3";

import "google/protobuf/empty.proto";

message RunningDetails {
  float distance = 1;
  float time = 2;
  uint32 heart_rate = 3;
  uint32 minutes_per_mile = 4;
}

message RunningMessageResponse {
  string message = 1;
}

message RunningMetrics {
  uint32 heart_rate = 1;
  float distance = 2;
  string time_stamp = 3;
}

service RunningMessage {
  rpc GetRunningMessage(RunningDetails) returns (RunningMessageResponse) {}
  rpc LogRunningMetrics(stream RunningMetrics) returns (google.protobuf.Empty) {}
  rpc GetRunningMetrics(google.protobuf.Empty) returns (stream RunningMetrics) {}
}